src/getters/awaitGraphBlock.ts

```diff
@@ -0,0 +1,32 @@
+import getGraphClient from '~/getters/getGraphClient'
+import {
+    GetMetadataDocument,
+    GetMetadataQuery,
+    GetMetadataQueryVariables,
+} from '~/generated/gql/network'
+
+export const awaitGraphBlock = (blockNumber: number): Promise<void> => {
+    return new Promise((resolve) => {
+        const interval = setInterval(async () => {
+            const client = getGraphClient()
+            const {
+                data: { _meta: meta },
+            } = await client.query<GetMetadataQuery, GetMetadataQueryVariables>({
+                query: GetMetadataDocument,
+            })
+
+            if (!!meta && meta.block.number >= blockNumber) {
+                clearInterval(interval)
+                resolve()
+            }
+
+            await client.clearStore()
+        }, 500)
+
+        setTimeout(() => {
+            // cancel polling after 15 seconds
+            clearInterval(interval)
+            resolve()
+        }, 15000)
```

Comment:
This feels like a timeout but it does resolve. Is that on purpose?

Reply:
to not keep the user too long - this awaiting for the graph sync is not the most crucial feature so my idea is that if it takes too long then we just ignore it and we let the user use the app normally